core/src/transport/mod.rs

```diff
@@ -18,13 +18,12 @@
 // FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 // DEALINGS IN THE SOFTWARE.
 
-//! Handles entering a connection with a peer.
+//! Connection-oriented communication channels.
 //!
-//! The main element of this module is the `Transport` trait. It is implemented on objects that
-//! allow dialing and listening.
-//!
-//! The rest of the module holds combinators that allow tweaking an implementation of `Transport`,
-//! combine multiple transports together, or combine a transport with an upgrade.
+//! The main entity of this module is the [`Transport`] trait, which provides an
```

Comment:
Does this automatically create a link to `Transport`? I know that such a feature was planned, but I don't know whether it is stable.

Reply:
Yes it does, but right now only when running `cargo doc` on nightly. It does not need a feature flag though. On stable these will just render as `[Foo]` instead of `Foo`, i.e. with brackets, but that seems harmless for the benefit of being forward-compatible. We could even generate the API docs using nightly before the RFC is stabilized, i.e. build the code with stable and only the docs with nightly to get the nicer API docs earlier.

Reply:
FYI this type of link in particular is referred to as [Implied Shortcut Reference Links](https://github.com/rust-lang/rfcs/blob/master/text/1946-intra-rustdoc-links.md#implied-shortcut-reference-links).